docs/concepts/what-is-an-activity-definition.md

```diff
@@ -9,8 +9,44 @@ tags:
 
 An Activity Definition is the code that defines the constraints of an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).
 
-We strongly recommend that you develop an Activity Definition in a language that has a corresponding Temporal SDK.
+Activity Definitions are referred to by their [Activity Type](/docs/concepts/what-is-an-activity-type).
+
+In Temporal documentation, the term 'Activity Definition' is used to refer to an Activity Function Definition——the method or function that is invoked for an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).
+
+Some SDKs refer to Activity Definitions as Activity Functions, Activity Interfaces, or Activity Methods.
+All of these terms refer to the source of an instance of an execution, with an Activity Interface being a collective group of functions.
+
+![Activity Definition](/diagrams/activity-definition.svg)
+
+#### Constraints
+
+Activity Definitions are executed as normal functions.
+
+In the event of failure, the function will begin at its initial state when retried (except when Activity Heartbeats are established).
+
+Therefore, an Activity Definition has no restrictions on the code it contains.
+
+#### Parameters
+
+An Activity Definition can support as many parameters as needed.
+
+All values passed through these parameters are recorded in the Workflow Execution's [Event History](/docs/concepts/what-is-an-event-history). Return values are also captured in the Event History for the calling Workflow Execution.
```

Comment:
```suggestion
All values passed through these parameters are recorded in the [Event History](/docs/concepts/what-is-an-event-history) of the Workflow Execution.
Return values are also captured in the Event History for the calling Workflow Execution.
```

First sentence tweaked to avoid possessive and thus provide easier translation later.